back-end/src/test/java/mine/is/gpu/gpuserver/domain/repository/GpuServerRepositoryTest.java

```diff
@@ -70,4 +75,49 @@ void delete() {
         Optional<GpuServer> actual = gpuServerRepository.findById(gpuServer.getId());
         assertThat(actual.isPresent()).isFalse();
     }
+
+    @DisplayName("Pageable를 파라미터로 GpuServer 정보를 가져온다")
+    @Test
+    void findAllByLabId() {
+        Lab lab = new Lab("lab");
+        labRepository.save(lab);
+
+        String baseName = "server";
+        IntStream.range(0, 30)
+                .mapToObj(i -> new GpuServer(baseName + i, false, 500L, 1024L, lab))
+                .forEach(gpuServer -> gpuServerRepository.save(gpuServer));
+
+        Pageable pageable = PageRequest.of(2, 3);
+
+        List<String> searchedNames = gpuServerRepository.findAllByLabId(lab.getId(), pageable).stream()
+                .map(GpuServer::getName)
+                .collect(Collectors.toList());
+
+        List<String> expectedNames = IntStream.range(0, pageable.getPageSize())
+                .mapToObj(i -> baseName + (pageable.getPageSize() * pageable.getPageNumber() + i))
+                .collect(Collectors.toList());
+
+        assertThat(expectedNames).usingRecursiveComparison().isEqualTo(searchedNames);
+    }
```

Comment:
```suggestion
        String[] searchedNames = gpuServerRepository.findAllByLabId(lab.getId(), pageable).stream()
                .map(GpuServer::getName)
                .toArray(String[]::new)

        List<String> expectedNames = IntStream.range(0, pageable.getPageSize())
                .mapToObj(i -> baseName + (pageable.getPageSize() * pageable.getPageNumber() + i))
                .collect(Collectors.toList());

        assertThat(expectedNames).containsExactly(searchedNames);
    }
```

>  `containsExactly()` 메소드의 사용은 어떨까요?

 List<>로 searchedNames를 구현할 수 있기는 한데, 내부 구현에서 다시 Array로 만드는 추가적인 과정이 포함되어 있어서, 제안은 Array로 진행해 봤습니다.
만약 순서가 상관없는 경우라면 `containsExactlyInAnyOrder()`를 사용해도 좋을 것 같구요!

`usingRecursiveComparison()` javadoc을 참고해보면, 이 방식은 List를 순회하는 용도보다는 객체 내부를 파고 들어가서 이 값들을 비교하는 로직에서 사용되는 것으로 생각했습니다!

물론 현재의 구현에서도, List String  객체 내부를 파고 들고 비교하기 때문에, 동작 상으로 동일한 로직이지만, containsExactly가 지금의 상황에 더 적합하다 생각해서 의견 남겨봐요!

아래의 테스트도 동일한 관점이라 생각해서 별도로 남기진 않겠습니다!

Reply:
1. searchedNames를 리스트 -> 배열
내부 구현에서 배열로 다시 바꾸는 로직이 포함된다고 해도, 저는 타입 안정성, 개발 편의, 기존 코드와의 통일을 위해 리스트 사용을 유지하겠습니다.  

2. 말씀대로 usingRecursiveComparison()이 불필요한 상황임을 확인하고 해당 함수 호출을 제거했습니다.

```
List<String> searchedNames = gpuServerRepository.findAllByLabId(lab.getId(), pageable).stream()
        .map(GpuServer::getName)
        .collect(Collectors.toList());

List<String> expectedNames = IntStream.range(0, pageable.getPageSize())
        .mapToObj(i -> baseName + (pageable.getPageSize() * pageable.getPageNumber() + i))
        .collect(Collectors.toList());

assertThat(expectedNames).isEqualTo(searchedNames);
```

Reply:
그냥 isEqaulTo를 사용해도 됬었군요 ㅎㅎ